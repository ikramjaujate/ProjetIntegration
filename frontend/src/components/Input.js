const Input = ( { type, name, min, max, placeholder, setFunc, pattern, title, step, idName} ) => {
    return (<input className="input_modif" id={idName} step={step} pattern={pattern} required title={title} type={type} name={name} minLength={min} maxLength={max} placeholder={placeholder} onChange={(e) => {
                                setFunc(e.target.value)
                            }}/>
    )
}

Input.defaultProps = {
    type : 'text',
    min : 4,
    max : 10,
    placeholder : 'texte'
}

export default Input