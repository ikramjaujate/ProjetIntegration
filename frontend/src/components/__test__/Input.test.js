import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16' ;
import 'jest-enzyme'
import 'jest-styled-components';
import Input from '../Input.js' ;
Enzyme.configure({ adapter: new Adapter() })
    

describe('CameraInfo', () => {

  it('Snapshot - renders correctement', () => {
    const snapshot = renderer.create(<Input type="text" name="camera" min={1} max={2} placeholder="text" setFunc="" pattern="text" title="text" step="text" idName="text"/>).toJSON();
    expect(snapshot).toMatchSnapshot();
  })

  it('Vérifier le caractère required du input.', () => {
    const wrapper = shallow(<Input required type="" name="" min="" max="" placeholder="" setFunc="" pattern="" title="" step="" idName=""/>)
    const paragraph = wrapper.find('input')
    expect(paragraph.at(0).props().required).toEqual(true);
  })
  it('Vérifier le type du input.', () => {
      const wrapper = shallow(<Input required type="text" name="" min="" max="" placeholder="" setFunc="" pattern="" title="" step="" idName=""/>)
      const paragraph = wrapper.find('input')
      expect(paragraph.at(0).props().type).toEqual("text");
    })
    it('Vérifier le name du input.', () => {
      const wrapper = shallow(<Input required type="" name="client" min="" max="" placeholder="" setFunc="" pattern="" title="" step="" idName=""/>)
      const paragraph = wrapper.find('input')
      expect(paragraph.at(0).props().name).toEqual("client");
    })
    it('Vérifier le placeholder du input.', () => {
      const wrapper = shallow(<Input required type="" name="" min="" max="" placeholder="text" setFunc="" pattern="" title="" step="" idName=""/>)
      const paragraph = wrapper.find('input')
      expect(paragraph.at(0).props().placeholder).toEqual("text");
    })
    it('Vérifier le min', () => {
      const wrapper = shallow(<Input min={2}/>)
      const paragraph = wrapper.find('input')
      expect(paragraph.at(0).props().minLength).toEqual(2);
    })
    it('Vérifier le max', () => {
      const wrapper = shallow(<Input max={8}/>)
      const paragraph = wrapper.find('input');
      expect(paragraph.at(0).props().maxLength).toEqual(8);
    })
    it('Vérifier le placeholder par default du input.', () => {
      const wrapper = shallow(<Input />)
      const paragraph = wrapper.find('input')
      expect(paragraph.at(0).props().placeholder).toEqual("texte");
    })
    it('Vérifier le text par default du input.', () => {
      const wrapper = shallow(<Input />)
      const paragraph = wrapper.find('input')
      expect(paragraph.at(0).props().type).toEqual("text");
    })
})