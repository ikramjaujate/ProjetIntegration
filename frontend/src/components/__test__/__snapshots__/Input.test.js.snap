// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraInfo Snapshot - renders correctement 1`] = `
<input
  className="input_modif"
  id="text"
  maxLength={2}
  minLength={1}
  name="camera"
  onChange={[Function]}
  pattern="text"
  placeholder="text"
  required={true}
  step="text"
  title="text"
  type="text"
/>
`;
